mmv1/third_party/validator/kms_crypto_key_iam.go

```diff
@@ -0,0 +1,71 @@
+package google
+
+import "fmt"
+
+func GetKmsCryptoKeyIamPolicyCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	return newKmsCryptoKeyIamAsset(d, config, expandIamPolicyBindings)
+}
+
+func GetKmsCryptoKeyIamBindingCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	return newKmsCryptoKeyIamAsset(d, config, expandIamRoleBindings)
+}
+
+func GetKmsCryptoKeyIamMemberCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	return newKmsCryptoKeyIamAsset(d, config, expandIamMemberBindings)
+}
+
+func MergeKmsCryptoKeyIamPolicy(existing, incoming Asset) Asset {
+	existing.IAMPolicy = incoming.IAMPolicy
+	return existing
+}
+
+func MergeKmsCryptoKeyIamBinding(existing, incoming Asset) Asset {
+	return mergeIamAssets(existing, incoming, mergeAuthoritativeBindings)
+}
+
+func MergeKmsCryptoKeyIamBindingDelete(existing, incoming Asset) Asset {
+	return mergeDeleteIamAssets(existing, incoming, mergeDeleteAuthoritativeBindings)
+}
+
+func MergeKmsCryptoKeyIamMember(existing, incoming Asset) Asset {
+	return mergeIamAssets(existing, incoming, mergeAdditiveBindings)
+}
+
+func MergeKmsCryptoKeyIamMemberDelete(existing, incoming Asset) Asset {
+	return mergeDeleteIamAssets(existing, incoming, mergeDeleteAdditiveBindings)
+}
+
+func newKmsCryptoKeyIamAsset(
+	d TerraformResourceData,
+	config *Config,
+	expandBindings func(d TerraformResourceData) ([]IAMBinding, error),
+) ([]Asset, error) {
+	bindings, err := expandBindings(d)
+	if err != nil {
+		return []Asset{}, fmt.Errorf("expanding bindings: %v", err)
+	}
+
+	name, err := assetName(d, config, "//cloudkms.googleapis.com/{{crypto_key_id}}")
```

Comment:
This should match what is in [terraform-google-conversion/google/kms_crypto_key.go](https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/9b13c676d981af90f533cd9526243355a4199f2a/google/kms_crypto_key.go#L24) - but it looks like that value is incorrect, since it uses kms.googleapis.com.

cloudkms.googleapis.com is definitely the [right base URL](https://cloud.google.com/asset-inventory/docs/supported-asset-types). However, I think there's probably going to need to be a sub-path that includes a key ring id similar to the kms_crypto_key's path.

You could create a crypto key in the cloud console and export it to find out what the path should look like, or find a reference in a more canonical codebase that lays out the expected format.

Reply:
Never mind, you're correct, `crypto_key_id` is the only option here: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_kms_crypto_key_iam#crypto_key_id